pkg/controller/livenessprobemapnodeprobe/probemapnodeprobe_utils.go

```diff
@@ -0,0 +1,223 @@
+package livenessprobemapnodeprobe
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+
+	alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	livenessprobeUtils "github.com/openkruise/kruise/pkg/util/livenessprobe"
+)
+
+func (r *ReconcileEnhancedLivenessProbeMapNodeProbe) delNodeProbeConfig(pod *v1.Pod) error {
+	return r.retryOnConflictDelNodeProbeConfig(pod)
+}
+
+func (r *ReconcileEnhancedLivenessProbeMapNodeProbe) addNodeProbeConfig(pod *v1.Pod) error {
+	plivenessProbeConfig, err := parseEnhancedLivenessProbeConfig(pod)
+	if err != nil {
+		return err
+	}
+	if pod.Spec.NodeName == "" {
+		klog.Warningf("No found pod node name, pod: %v/%v", pod.Namespace, pod.Name)
+		return nil
+	}
+
+	npp := appsv1alpha1.NodePodProbe{}
+	err = r.Get(context.TODO(), types.NamespacedName{Name: pod.Spec.NodeName}, &npp)
+	if err == nil {
+		return r.retryOnConflictAddNodeProbeConfig(pod, plivenessProbeConfig)
+	}
+	if !errors.IsNotFound(err) {
+		klog.Errorf("Failed to get node pod probe for pod: %s/%s, err: %v", pod.Namespace, pod.Name, err)
+		return err
+	}
+	newNppGenerated, err := r.generateNewNodePodProbe(pod)
```

Comment:
pass plivenessProbeConfig to generateNewNodePodProbe, it is not necessary to parse the liveness probe twice

Reply:
func generateNewNodePodProbe(...) has been updated for using the podLivenessProbe from annotation as the paramater.